gmloader/main.cpp

```diff
@@ -109,6 +110,45 @@ int main(int argc, char *argv[])
 
     }
 
+    char platform_ov[32];
+    strncpy(platform_ov,gmloader_config.force_platform.c_str(),sizeof(platform_ov)-1);
+    platform_ov[31] = '\0';
+    
+    if (platform_ov) {
+        for (int i = 0; platform_ov[i] != '\0'; i++)
+            platform_ov[i] = tolower(platform_ov[i]);
+
+        if(strcmp(platform_ov, "os_unknown") == 0){
+            FORCE_PLATFORM = os_unknown;
+        }else if(strcmp(platform_ov, "os_windows") == 0){
+            FORCE_PLATFORM = os_windows;
+        }else if(strcmp(platform_ov, "os_macosx") == 0){
+            FORCE_PLATFORM = os_macosx;
+        }else if(strcmp(platform_ov, "os_ios") == 0){
+            FORCE_PLATFORM = os_ios;
+        }else if(strcmp(platform_ov, "os_android") == 0){
+            FORCE_PLATFORM = os_android;
+        }else if(strcmp(platform_ov, "os_linux") == 0){
+            FORCE_PLATFORM = os_linux;
+        }else if(strcmp(platform_ov, "os_psvita") == 0){
+            FORCE_PLATFORM = os_psvita;
+        }else if(strcmp(platform_ov, "os_ps4") == 0){
+            FORCE_PLATFORM = os_ps4;
+        }else if(strcmp(platform_ov, "os_xboxone") == 0){
+            FORCE_PLATFORM = os_xboxone;
+        }else if(strcmp(platform_ov, "os_tvos") == 0){
+            FORCE_PLATFORM = os_tvos;
+        }else if(strcmp(platform_ov, "os_switch") == 0){
+            FORCE_PLATFORM = os_switch;
+        }else{
+            fatal_error("Unexpected platform '%s'.\n", platform_ov);
```

Comment:
If we force a fatal error here does it mean the application stops? Shouldn't we just default to os_unknown or fallback to os_android?

Reply:
in the current implementation fatal_error doesn't make the application to stop. Also it fallbacks to `os_unkown`.
```bash
./gmloadernext.aarch64 -c /devel/gmloader-next/configuration.gmld
Loading config file (/devel/gmloader-next/configuration.gmld)
	save_dir = gamedata
	apk_path = my_game.apk
	show_cursor = 0
	disable_controller = 0
	force_platform = os_wind0ws
gmloader/main.cpp:144: Unexpected platform 'os_wind0ws'.
os_type = os_unknown
gmloader/main.cpp:160: Failure opening APK '/devel/gmloader-next/build/aarch64-linux-gnu/gmloader/my_game.apk': No such file
```

However, maybe using `warning()` will be appropriate.